packages/react-devtools-shared/src/backend/utils.js

```diff
@@ -52,3 +58,16 @@ export function copyWithSet(
   updated[key] = copyWithSet(obj[key], path, value, index + 1);
   return updated;
 }
+
+export function safeSerialize(data: any): string {
+  const cache = new Set();
+  return JSON.stringify(data, (key, value) => {
```

Comment:
Can you please explain what makes this safe?

Reply:
Maybe "safe" is a poor choice of words. This function protects against circular references, which would otherwise cause `JSON.stringify` to throw.

Reply:
Ah okay, I was wondering in cases of using Symbols or functions etc, would it be safe for those too? Naming things is hard :/

Reply:
It would be "safe" in that case, it would just not copy anything meaningful to the clipboard. (If a value can't be serialized, `JSON.stringify` would just return `undefined`.)

I renamed "safeSerialize" to "serializeToString" though and added an inline comment about its purpose.

Reply:
That’s a cool trick @bvaughn 👌🏽